### PR TITLE
Fix resource leaks in `integrationTestJdbc`

### PR DESCRIPTION
**Pull request by komamitsu**
## Description

I found some resource leaks including potential ones while executing `integrationTestJdbc`.  Resource leaks can cause unexpected exceptions (e.g., no available connection) and make it difficult to find actual problems of the implementation. This PR is to fix the resource leaks.

## Related issues and/or PRs

None

## Changes made

This PR contains the following changes:
- Improve the error handling of the clean-up operations in `@AfterAll` and `@AfterEach` annotated methods
- Release the resources that are not released

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

None

## Release notes

N/A

### CODE REVIEW

File: integration-test/build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ dependencies {
     implementation project(':core')
     implementation project(':schema-loader')
     implementation "com.google.guava:guava:${guavaVersion}"
+    implementation "org.slf4j:slf4j-api:${slf4jVersion}"
```
Comment (brfrn169) [inline integration-test/build.gradle:12]: Maybe we also need to add some `slf4j` logger implementation like `org.slf4j:slf4j-simple`?

**Review by brfrn169 — COMMENTED** [review]
Overall LGTM. Left a comment. Please take a look when you have time!

File: integration-test/build.gradle (hunk 1 above)
Comment (komamitsu) [reply integration-test/build.gradle:12]: Good catch! I think so. Actually I saw some log messages, but I guess `org.slf4j:slf4j-simple` came from `schema-loader`... I'll explicitly add the dependency here.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**